File: Array/1394. Find Lucky Integer in an Array.cpp
class Solution {
public:
    int findLucky(vector<int>& arr) {
        const auto& count = counter(arr);
        int result = -1;
        for (const auto& [k, v] : count) {
        //or for(auto [k,v ]: cnt)
            if (k == v) {
                result = max(result, k);
            }
        }
        return result;
    }

private:
    unordered_map<int, int> counter(const vector<int>& arr) {
        unordered_map<int, int> count;
        for (const auto& i : arr) {
            ++count[i];
        }
        return count;
    }
};

File: Array/1394. Find Lucky Integer in an Array.py
class Solution:
    def findLucky(self, arr: List[int]) -> int:
        c = collections.Counter(arr)
        res = -1
        for i, v in c.items():
            if i == v:
                res = max(res, i)
        if res != -1:
            return res
        return -1